vars/createSha512Checksums.groovy

```diff
@@ -0,0 +1,26 @@
+Closure call() {
+    allowedFileTypes = [".tar.gz", ".zip"]
+
+    return { argsMap -> body: {
+
+        final foundFiles = sh(script: "find $argsMap.artifactPath -type f", returnStdout: true).split()
+
+        for (file in foundFiles) {
+            acceptTypeFound = false
+            for (fileType in allowedFileTypes) {
+                if (file.endsWith(fileType)) {
+                    echo("Creating sha for ${file}")
+                    final sha512 = sh(script: "sha512sum ${file}", returnStdout: true).split()
+                    // Reading the first index of array since sha512sum return [shasum, filename] as output
+                    writeFile file: "${file}.sha512", text: "${sha512[0]}"
+                    acceptTypeFound = true
+                    break
```

Comment:
Passing the allowed file types into `find` would simplify this a lot. See https://stackoverflow.com/questions/2621513/how-to-use-find-command-to-find-all-files-with-extensions-from-list.

Reply:
I agree. However, I think the current implementation also is good, since it mentions the files that are skipped. It helped to keep a check if anything is skipped when we ran the job.

Reply:
The current implementation is twice as long and has poor separation of concerns (finding what to sign and signing it). Compare with:

```
final foundFiles = sh(.... allowedFileTypes)

for (file in foundFiles) {
  echo "Creating sha ..."
  final sha512 = 
  writeFile
}
```

Reply:
How will we know the files that were skipped with this implementation?

Reply:
You don't need to know what's skipped, you only need to know which files to sign.